include/toml++/impl/source_region.hpp

```diff
@@ -217,6 +217,70 @@ TOML_NAMESPACE_START
 			return lhs;
 		}
 	};
+
+	/// \brief	Returns the line at the specified line number, from the specified document.
+	///
+	/// \detail \cpp
+	/// auto doc = "alpha = 1\nbeta = 2"sv;
+	/// auto second_line = toml::get_line(doc, 2);
+	///
+	/// std::cout << "The second line says \"" << second_line << "\"\n";
+	/// \ecpp
+	///
+	/// \out
+	/// The second line says "beta = 2"
+	/// \eout
+	///
+	/// \param 	doc	The document.
+	/// \param 	line_num	The line number (1-based).
+	///
+	/// \returns	The specified line, excluding any possible trailing carriage return or line feed character.
+	/// \remarks Returns an empty string_view when there is no line at the specified line number, in the specified document.
+	inline std::string_view get_line(std::string_view doc, source_index line_num)
+	{
+		if (line_num == 0)
+		{
+			// Invalid line number. Should be greater than zero.
+			return {};
+		}
+
+		// The position of the first character of the specified line.
+		const auto begin_of_line = [doc, line_num]() -> std::size_t
+		{
+			if (line_num == 1)
+			{
+				return 0;
+			}
+
+			const auto num_chars = doc.size();
+			std::size_t current_line_num{ 1 };
+
+			for (std::size_t i{}; i < num_chars; ++i)
+			{
+				if (doc[i] == '\n')
+				{
+					++current_line_num;
+
+					if (current_line_num == line_num)
+					{
+						return i + 1;
+					}
+				}
+			}
+			return std::string_view::npos;
+		}();
+
+		if (begin_of_line >= doc.size())
+		{
+			return {};
+		}
+
+		const auto end_of_line = doc.find_first_of("\r\n", begin_of_line);
```

Comment:
Also I think this should just be `\n`:

```suggestion
		const auto end_of_line = doc.find_first_of("\n", begin_of_line);
```

Reply:
The TOML spec would regard that as being just one line, so the same thing it would return for `"embedded X here"`.

> as it seemed like an exotic and irrelevant scenario to me

For your use-case, yes. But if it's in the library proper, I don't get the luxury of thinking like that unfortunately 😅